app/presenters/expense_presenter.rb

```diff
@@ -76,4 +76,22 @@ def show_map_link?
       true
     end
   end
+
+  def state
+    if expense.mileage_rate_id&.eql?(1) &&
+       expense.calculated_distance.present? &&
+       (expense.distance <= expense.calculated_distance)
```

Comment:
does this handle the previously encountered nil value comparison problems?

Reply:
perhaps also a method just for `expense.calculated_distance.present? && expense.distance <= expense.calculated_distance`
like `distance_acceptable?` as it is used below too.